src/lib.rs

```diff
@@ -33,6 +37,51 @@ pub fn links(bytes: &[u8], links: &mut impl Extend<Cid>) -> Result<(), Error> {
     Ok(())
 }
 
+/// DAG-PB implementation of ipld-core's `Codec` trait.
+pub struct DagPbCodec;
+
+impl Codec<Ipld> for DagPbCodec {
+    const CODE: u64 = 0x70;
+    type Error = Error;
+
+    fn decode<R: BufRead>(mut reader: R) -> Result<Ipld, Self::Error> {
+        let mut bytes = Vec::new();
+        reader.read_to_end(&mut bytes)?;
+        crate::to_ipld(&bytes)
+    }
+
+    fn encode<W: Write>(mut writer: W, data: &Ipld) -> Result<(), Self::Error> {
+        let bytes = crate::from_ipld(data)?;
+        Ok(writer.write_all(&bytes)?)
+    }
+}
+
+impl Links for DagPbCodec {
+    type LinksError = Error;
+
+    fn links(data: &[u8]) -> Result<impl Iterator<Item = Cid>, Self::LinksError> {
+        let mut links = Vec::new();
+        crate::links(data, &mut links)?;
```

Comment:
out of interest: why does this take a `&mut links` as an argument rather than just returning one? is there benefit in putting ownership into the caller, or are there uses here where we actually want to extending an existing vector?

Reply:
It means:

1. You can use any collection you want, as long as it implements `Extend`.
2. You can extract links from multiple blocks into the same vector without copying.

Although, in this case, I'm not sure why we don't just return an iterator and/or vector, given that we already have one.

Reply:
It's built on existing code and I didn't want to break peoples code only because I can (I hope the DAG-PB usage is minimal anyway).